src/services/users.service.js

```diff
@@ -0,0 +1,59 @@
+import { ApiError } from '../exeptions/api.error.js';
+import { User } from './../models/index.js';
+import { v4 as uuidv4 } from 'uuid';
+import { emailService } from './index.js';
+
+export const usersService = {
+  async register({ email, password, name }) {
+    const isExist = !!(await this.getOne({ email }));
+
+    if (isExist) {
+      throw ApiError.badRequest('User already exist', {
+        email: 'User already exist',
+      });
+    }
+
+    const activationToken = uuidv4();
+
+    await User.create({
+      email,
+      password,
+      name,
+      activationToken,
+    });
+
+    await emailService.sendActivationEmail(email, activationToken);
+  },
+
+  async getOne(params) {
+    const user = await User.findOne({
+      where: {
+        ...params,
+      },
+    });
+
+    return user;
+  },
+
+  async update(query, newParams) {
+    const [affectedRows, [updatedUser]] = await User.update(
```

Comment:
The `update` method uses `returning: true`, which is specific to certain SQL dialects. Ensure that your database supports this option, or handle the response accordingly.